contracts/secret-storage/SecretStore.sol

```diff
@@ -477,211 +516,185 @@ contract SecretStore is
-        _markDeadUpdate(_enclaveAddress, _currentCheckTimestamp, _markAliveTimeout, _storageOccupied, _recipient);
+    /**
+     * @notice Slashes the stake of an executor node.
+     * @dev Can only be called by an account with the `JOBS_ROLE`. This function
+     *      triggers a slashing penalty on the specified executor node.
+     * @param _enclaveAddress The address of the executor enclave to be slashed.
+     * @return The amount of stake that was slashed from the executor node.
+     */
+    function slashExecutor(address _enclaveAddress) external onlyRole(JOBS_ROLE) returns (uint256) {
+        return _slashExecutor(_enclaveAddress, _msgSender());
     }
 
-    function secretTerminationUpdate(
-        address _enclaveAddress,
-        uint256 _sizeLimit,
-        uint256 _secretId
-    ) external onlyRole(SECRET_MANAGER_ROLE) {
-        _secretTerminationUpdate(_enclaveAddress, _sizeLimit, _secretId);
+    function updateTreeState(address _enclaveAddress) external {
+        _updateTreeState(_enclaveAddress);
     }
 
-    function getSecretStoreOwner(address _enclaveAddress) external view returns (address) {
-        return secretStorage[_enclaveAddress].owner;
+    function selectNodes(
+        uint8 _env,
+        uint256 _noOfNodesToSelect
+    ) external view isValidEnv(_env) returns (address[] memory selectedNodes) {
+        return _selectNodes(_env, _noOfNodesToSelect);
     }
 
-    function getSecretStoreLastAliveTimestamp(address _enclaveAddress) external view returns (uint256) {
-        return secretStorage[_enclaveAddress].lastAliveTimestamp;
+    function increaseReputation(address _enclaveAddress, uint256 _value) external onlyRole(JOBS_ROLE) {
+        executors[_enclaveAddress].reputation += _value;
     }
 
-    function getSecretStoreDeadTimestamp(address _enclaveAddress) external view returns (uint256) {
-        return secretStorage[_enclaveAddress].deadTimestamp;
+    function decreaseReputation(address _enclaveAddress, uint256 _value) external onlyRole(JOBS_ROLE) {
+        executors[_enclaveAddress].reputation -= _value;
     }
 
-    function getStoreAckSecretIds(address _enclaveAddress) external view returns (uint256[] memory) {
-        return secretStorage[_enclaveAddress].ackSecretIds;
+    //---------------------------------- external functions end ------------------------------------//
+
+    //---------------------------------- JobsRole functions end -------------------------------------//
+
+    //----------------------------- SecretManagerRole functions start --------------------------------//
+
+    //-------------------------------- external functions start ----------------------------------//
+
+    function getSecretStoreOwner(address _enclaveAddress) external view returns (address) {
+        return executors[_enclaveAddress].owner;
     }
 
-    function addAckSecretIdToStore(
-        address _enclaveAddress,
-        uint256 _ackSecretId
+    function deleteIfPresent(
```

Comment:
need to make sure, even if storage is full, executor should be able to get selected for jobs.